src/com/facebook/buck/shell/WorkerProcessProtocolZero.java

```diff
@@ -239,6 +239,8 @@ public void close() throws IOException {
       processStdinWriter.close();
       processStdoutReader.endArray();
       processStdoutReader.close();
+    } catch (IllegalStateException ex) {
+      throw new IOException(ex);
```

Comment:
Opened #960 and reverted this. Originally the callers only caught `IOException` and this seemed IO-ish, but it's probably more error-proof to just catch `Exception` in callers.